vqfx/Makefile

```diff
@@ -1,16 +1,35 @@
 VENDOR=Juniper
 NAME=vQFX
-IMAGE_FORMAT=vmdk
-IMAGE_GLOB=*.vmdk
+IMAGE_FORMAT=qcow2
+IMAGE_GLOB=*.qcow2
 
-# match versions like:
-# vqfx10k-re-15.1X53-D60.vmdk
-VERSION=$(shell echo $(IMAGE) | sed -e 's/^vqfx10k-re-\([0-9][0-9]\.[0-9][A-Z][0-9]\+\(\.[0-9]\+\)\?\(-D[0-9]\+\)\)[^0-9].*$$/\1/')
+# New vqfx are named: vqfx-19.4R1.10-re-qemu.qcow2
+VERSION=$(shell echo $(IMAGE) |  sed -e 's/^vqfx-//'|sed -e 's/-re-qemu.qcow2//' )
+# Old QFX are named: vqfx10k-re-15.1X53-D60.vmdk, and vqfx10k-re-18.4R1.8.vmdk and vqfx10k-pfe-18.4R1.8.vmdk so we need that version number extracted to convert them. 
+VMDK_VERSION:=$(shell ls *-re-*.vmdk | sed -re 's/vqfx10k-re-([^;]*)\.vmdk.*$$/\1/')
+
+PFE_BASE_VERSION=$(shell echo $VERSION | sed -e s'/.*//')
+PFE_IMAGE=$(shell ls vqfx-$(PFE_BASE_VERSION)*-pfe-qemu.qcow*)
 
 -include ../makefile-sanity.include
 -include ../makefile.include
 
-# vqfx10k-pfe-20160609-2.vmdk
+format-legacy-images:
+	@if ls *.vmdk; then echo "VMDKs exist, converting them to qcow format"; qemu-img convert -f vmdk -O qcow2 *-re-*.vmdk vqfx-$(VMDK_VERSION)-re-qemu.qcow2 && qemu-img convert -f vmdk -O qcow *-pfe-*.vmdk vqfx-$(VMDK_VERSION)-pfe-qemu.qcow; echo "VMDKs have been converted"; fi
+
 # TODO: we should make sure we only copy one PFE image (the latest?), in case there are many
 docker-pre-build:
-	cp vqfx10k-pfe*.vmdk docker/
+
+	echo "pfe     $(PFE_IMAGE)"
```

Comment:
why do we copy only the pfe image? what happens with re?

Reply:
I am not entirely sure as that came from the PR#234 in the upstream repo, however, I _think_ the RE gets copied in and added to the image at [line 9](https://github.com/hellt/vrnetlab/blob/7542cf1cceb266935a3f59dbbc2bc63dcae1639c/makefile.include#L9) of the `makefile.include` or possibly during [line 34](https://github.com/hellt/vrnetlab/blob/7542cf1cceb266935a3f59dbbc2bc63dcae1639c/vqfx/Makefile#L34) of this Makefile, which calls the make target on [line 21](https://github.com/hellt/vrnetlab/blob/7542cf1cceb266935a3f59dbbc2bc63dcae1639c/makefile.include#L21) of the `makefile.include`.  This entire step of copying the PFE might actually be redundant and not needed. Let me test without it and see what happens.

Reply:
Okay, so it is not redundant and is definitely needed. Without having dug too deep, I'm guessing that the built-in stuff only handles 1 disk image, so this is necessary to get the PFE into docker.